operator/src/main/java/oracle/kubernetes/operator/DomainEventWatcher.java

```diff
@@ -0,0 +1,61 @@
+// Copyright (c) 2021, Oracle and/or its affiliates.
+// Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl.
+
+package oracle.kubernetes.operator;
+
+import java.util.concurrent.ThreadFactory;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import io.kubernetes.client.openapi.ApiException;
+import io.kubernetes.client.openapi.models.V1Event;
+import io.kubernetes.client.util.Watchable;
+import oracle.kubernetes.operator.TuningParameters.WatchTuning;
+import oracle.kubernetes.operator.builders.WatchBuilder;
+import oracle.kubernetes.operator.watcher.WatchListener;
+
+import static oracle.kubernetes.operator.ProcessingConstants.DOMAIN_EVENT_LABEL_FILTER;
+
+/**
+ * This class handles Domain Event watching. It receives event notifications and sends them into the operator
+ * for processing.
+ */
+public class DomainEventWatcher extends EventWatcher {
+
+  private DomainEventWatcher(
+        String ns,
+        String initialResourceVersion,
+        WatchTuning tuning,
+        WatchListener<V1Event> listener,
+        AtomicBoolean isStopping) {
+    super(ns, initialResourceVersion, tuning, listener, isStopping);
+  }
+
+  /**
+   * Create and start a new DomainEventWatcher.
+   * @param factory thread factory to use for this watcher's threads
+   * @param ns namespace
+   * @param initialResourceVersion the oldest version to return for this watch
+   * @param tuning Watch tuning parameters
+   * @param listener a listener to which to dispatch watch events
+   * @param isStopping an atomic boolean to watch to determine when to stop the watcher
+   * @return the domain watcher
+   */
+  public static DomainEventWatcher create(
+      ThreadFactory factory,
+      String ns,
+      String initialResourceVersion,
+      WatchTuning tuning,
+      WatchListener<V1Event> listener,
+      AtomicBoolean isStopping) {
+    DomainEventWatcher watcher =
+        new DomainEventWatcher(ns, initialResourceVersion, tuning, listener, isStopping);
+    watcher.start(factory);
+    return watcher;
+  }
+
+  @Override
```

Comment:
This overrides the corresponding method in EventWatcher, but specifies a _different_ label selector. That suggests that, although it purports to be a specialization of EventWatcher, it is looking for a different set of events, not simply a subset of them. 

See https://stackify.com/solid-design-liskov-substitution-principle/

It also notable that neither this method nor the one it overrides are unit-tested. Tests might have revealed this issue.

Reply:
Good point. DomainEventWatcher should not extend EventWatcher - I'll change that.  
EventWatcher is tested in EventProcessingTest, but may not have sufficient coverage. I'll see if I can add more unit tests to DomainEventWatcher itself, and EventWatcher.